model/modeldecoder/generator/generator.go

```diff
@@ -100,34 +96,26 @@ func NewGenerator(importPath string, pkg string, typPath string,
 
 // Generate writes generated methods to the buffer
 func (g *Generator) Generate() (bytes.Buffer, error) {
-	// run generator code, writing code to the buffer
-	// and collect required imports
-	for _, rootObj := range g.rootObjs {
-		if err := g.generate(rootObj, ""); err != nil {
-			return g.buf, err
-		}
-	}
-
-	// write package, edit note and imports to temporary buffer
-	var buf bytes.Buffer
-	fmt.Fprintf(&buf, `
+	fmt.Fprintf(&g.buf, `
 // Code generated by "modeldecoder/generator". DO NOT EDIT.
 
 package %s
 
 import (
 	"fmt"
+	"encoding/json"
+	"github.com/pkg/errors"
+	"unicode/utf8"
```

Comment:
I dediced to not pass in the imports but have them hardcoded, as they depend on generator internas and are not known to the caller.